rust-k256/src/randomizedsigner.rs

```diff
@@ -0,0 +1,110 @@
+use super::{
+    CryptoRngCore, NonZeroScalar, PlumeSignature, PlumeSignatureV1Fields, ProjectivePoint,
+    SecretKey, DST,
+};
+use k256::{
+    elliptic_curve::{
+        hash2curve::{ExpandMsgXmd, GroupDigest},
+        point::NonIdentity,
+        sec1::ToEncodedPoint,
+    },
+    sha2::{Digest, Sha256},
+    Secp256k1,
+};
+// Removed `pub` from this, since it's only interested to those who already imported `signature`
+use signature::{Error, RandomizedSigner};
+
+/// `PlumeSigner` is a `struct` that contains a reference to a secret key and a
+/// boolean defining output [`PlumeSignature`] variant. It implements the
+/// `RandomizedSigner` trait to generate signatures using the provided secret
+/// key. The struct is generic over the lifetime of the secret key reference
+/// so that the key can be borrowed immutably.
+pub struct PlumeSigner<'signing> {
+    /// The secret key to use for signing. This is borrowed immutably.
+    secret_key: &'signing SecretKey,
+    /// Whether to generate a PlumeSignature V1 (true) or PlumeSignature V2 (false).
+    ///
+    /// `bool` is fine to use here since the choice affects only the hashing which doesn't
+    /// involve the key material, and distinguishing on it doesn't look possible
+    // Since #lastoponsecret seems to me indistinguishible between variants here's `bool` is used instead of `subtle`
+    pub v1: bool,
+}
+impl<'signing> PlumeSigner<'signing> {
+    /// Creates a new `PlumeSigner` instance with the given secret key and signature
+    /// variant.
+    pub fn new(secret_key: &SecretKey, v1: bool) -> PlumeSigner {
+        PlumeSigner { secret_key, v1 }
+    }
+}
+impl<'signing> RandomizedSigner<PlumeSignature> for PlumeSigner<'signing> {
+    fn try_sign_with_rng(
```

Comment:
Is there a reason the user has to pass in the RNG? Is it standard to have a version in which the RNG is determined inside of the function itself? Maybe I'm not as familiar with how RNG is treated in rust.

Reply:
This particular part is conforming to RustCrypto, outside of the module I preserved choice of the CRNG as they're sensitive thing, and it's a good practice to offer downstream to use whatever they see fit (according to the trait bounds). 

On the other hand if you feel like simplifying it further I see no problem to just hard-code the system CRNG like the example shows inputting it. I guess downstream wouldn't have the control for the thread in which that will be running. But that path/choice is mainly for no-std (embedded) environment where instantiating the CRNG is custom to the platform. Anyway downstream would be able to sign with the PlumeSigner which offers the choice.

Reply:
What if we exposd a fn for signing with a preset RNG as well, kind of like a default arg?

Reply:
That's exactly what I'm talking about. Just initializing the system RNG in `sign_v1` leaving `PlumeSigner` for those who need detailed control.